src/openms/source/FORMAT/MSPGenericFile.cpp

```diff
@@ -131,20 +132,42 @@ namespace OpenMS
       {
         spectrum.setRT(std::stod(m[1]));
       }
-      // Specific case of NIST's exported msp
+      //CAS# NIST#
       else if (boost::regex_search(line, m, re_cas_nist))
       {
         // OPENMS_LOG_DEBUG << "CAS#: " << m[1] << "; NIST#: " << m[2] << "\n";
         spectrum.setMetaValue(String("CAS#"), String(m[1]));
         spectrum.setMetaValue(String("NIST#"), String(m[2]));
       }
-      // Other metadata
+      // Other metadata, needs to be last, matches everything
       else if (boost::regex_search(line, m, re_metadatum))
       {
         // OPENMS_LOG_DEBUG << m[1] << m[2] << "\n";
         spectrum.setMetaValue(String(m[1]), String(m[2]));
       }
     }
+    // ensure correct meta value names for MetaboliteSpectralMatcher
+    spectrum.setMetaValue("Metabolite_Name", spectrum.getName());
```

Comment:
Put the metavalue names to Constants.h?

Reply:
same for MGF

Reply:
Agreed, it would make sense to put all the meta values relevant for MetaboliteSpectralMatching into Constants.h